Docs/ 1158 Combobox finetune documentation (#1647)
sl-design-system/components: website/src/categories/components/combobox/usage.md

---
title: Combobox usage
tags: usage
eleventyNavigation:
  parent: Combobox
  key: ComboboxUsage
---

<section class="no-heading">

<div class="ds-example">

> <sl-combobox multiple value='["0","1"]'>

Reviewer: Is it ok that in the example, when we open the page, the combobox is already focused and opened?

<img width="937" alt="image" src="https://github.com/user-attachments/assets/f0cf517e-a510-47ea-971a-3bb06f15d0b7" />

Reviewer 2: No, that's a bug i intend to fix (in another branch).

  <sl-listbox>
    <sl-option value="0">Mathematics</sl-option>
    <sl-option value="1">Geography</sl-option>
    <sl-option value="2">Physics</sl-option>
    <sl-option value="3">History</sl-option>
  </sl-listbox>
</sl-combobox>

</div>

<div class="ds-code">

  ```html
  <sl-combobox multiple value='["0","1"]'>
    <sl-listbox>
      <sl-option value="0">Mathematics</sl-option>
      <sl-option value="1">Geography</sl-option>
      <sl-option value="2">Physics</sl-option>
      <sl-option value="3">History</sl-option>
    </sl-listbox>
  </sl-combobox>
  ```

</div>

</section>

<section>

## When to Use
Use a Combobox to help users easily search, filter, and select options from long list of options.

> ### **Large Datasets**

Reviewer 3: ```suggestion
### Large Datasets
```

Comboboxes are a better choice than checkbox or radio button groups when dealing with more than 15 options. Unlike static groups that can overwhelm users with lengthy lists, Comboboxes save space and simplify the interface by enabling search and filtering. This makes it easier and faster for users to find their desired option, especially in large datasets like cities, products, or categories, without scrolling through extensive lists.

### Multiple Choices

> Our comboboxes come with multi-selection capabilities provide a clear overview of selected options by displaying them within a tag list. This is ideal when users need to choose multiple items, this ensures selections are visible and easy to manage, allowing users to remove or modify choices effortlessly while keeping the interface organized and intuitive.

Reviewer 3: ```suggestion
Our combobox comes with multi-selection capabilities that provide a clear overview of selected options by displaying them within a tag list in the input field of the combobox. This is ideal when users need to choose multiple items, as it ensures selections are visible and easy to manage, allowing users to remove or modify choices effortlessly while keeping the interface organized and intuitive.
```

</section>

<section>

## When Not to Use

> Avoid a Combobox when users benefit more from seeing all options at once for quick selection.

Reviewer 3: ```suggestion
Avoid a Combobox when users would benefit more from seeing all options at once for quick selection.
```

### Fewer Options**

> For list of options small than 15 choices, use [Checkbox](/categories/components/checkbox/usage) or [Radio Button](/categories/components/radio-group/usage) instead of a Combobox. These components present all options visible, making them quicker and more intuitive for users to scan and select without additional interactions.

Reviewer 3: ```suggestion
For list of options small than 15 choices, use [Checkbox](/categories/components/checkbox/usage) or [Radio Button](/categories/components/radio-group/usage) instead of a Combobox. These components present all options visible without it needing a user interaction first, making them quicker and more intuitive for users to scan and select without additional interactions.
```

> </section>

Author: I added a new point for this comment. We will also cover this on the selection patterns page.

> ## Short Lists

Reviewer: ```suggestion
### Short Lists
```

<section>

## Anatomy

|Item|Name| Description | Optional|
|-|-|-|-|
|1|Input Field |Acts as the primary area for user interaction. Users can type to search, or view the selected options in this field. |no|
|2|Indicator |A visual indicator showing the current state of the dropdown (expanded or collapsed) and guiding users to interact with the Combobox. |no|
|2|Tag List |Displays the chosen options within the input field or as separate tags, ensuring clarity about the current selection. |Yes|
|3|Menu |Displays available options dynamically. These can be split in groups to make navigation through long lists easier, and the choices will be display at the top. |no|

{.ds-table .ds-table-align-top}

</section>

<section>

## Variants

> Contextual Menu comes in two versions, each suited for specific situations:

Reviewer 3: ```suggestion
Combobox comes in two versions, each suited for specific situations:
```

> |3|Menu |Displays available options dynamically. These can be split in groups to make navigation through long lists easier, and the choices will be display at the top. |no|

Reviewer: ```suggestion
|3|Menu |Displays available options dynamically. These can be split in groups to make navigation through long lists easier, and the choices will be displayed at the top. |no|
```

<section>

## Behaviors
Let's explore the behavior of the Combobox

### Grouping Choices
Provides organizational structure for selected items by categorizing them under specific groups or labels. This behavior is especially beneficial in complex datasets where selections belong to distinct categories.

### Stacked Selections
Displays multiple selected items as a visually stacked list at the top of the dropdow list or within tag list. This approach keeps the interface tidy and ensures users can quickly review or edit their choices without scrolling through a long dropdown.

### Dynamic Filtering
Filters the options list in real time based on user input, displaying only relevant matches. This functionality enhances usability, particularly for lists with many similar or closely related items.

### Autocomplete Suggestions
The Combobox provides real-time autocomplete suggestions as users type. This behavior helps narrow down options by displaying only relevant matches, making the selection process faster and more efficient, especially in large datasets. Users can select a suggestion or continue typing to refine their input further.

</section>

<section>

## Related components

> 

Reviewer: Maybe here should be a `tag` component mentioned as well?

</section>

<section>

## When Not to Use
Avoid a Combobox when users would benefit more from seeing all options at once for quick selection.

### Fewer Options
For list of options small than 15 choices, use [Checkbox](/categories/components/checkbox/usage) or [Radio Button](/categories/components/radio-group/usage) instead of a Combobox. These components present all options visible, making them quicker and more intuitive for users to scan and select without additional interactions.

## Short Lists
Use a Select instead of a Combobox when the list of options contains fewer than 15 items. The simplicity of a [Select](/categories/components/select/usage) allows users to see all options at once, making it quicker and more straightforward for them to make a choice without the need for search or filtering.

> 

Reviewer 3: It isn't really clear when you should use a list of checkboxes and when a select. They both say "use this when there's fewer than 15 options."
Plus the rest of the text doesn't make sense for a select you don't see "all options at once" in a select with 15 options...
